starter-code/src/functions-and-arrays.js

```diff
@@ -11,13 +22,65 @@ var words = [
   'crackpot'
 ];
 
-// Calculating a Sum
+function findLongestWord(array){
+  for (var longestWord = '', i = 0; i < array.length; i++) {
+    if (array[i].length > longestWord.length) {
+      longestWord = array[i];
+    }    
+  }
+  return longestWord;
+}
+console.log(findLongestWord(words));
+
+
+
 
+
+
+// Calculating a Sum without method
 var numbers = [6, 12, 1, 18, 13, 16, 2, 1, 8, 10];
 
-// Calculate the Average
+function sum(array){
+  for (var suma= 0, i = 0; i < array.length; i++) {
+    suma+= array[i];   
+  }
+  return suma;
+}
+console.log(sum(numbers));
+
+
+//// Calculating a Sum with reduce
+var suma = numbers.reduce(function(acc, curr, i, arr){  
+  return acc + curr;
+}, 0);
+console.log(suma);
+
+
+//Calculating sum with reduce 2
+function sum(array1){
+  return numbers.reduce((acc, curr, i, arr) => acc + curr);
+}
+console.log(sum(numbers));
+
+
+
+
+
+
+// // Calculate the Average
+var numbers = [2, 6, 9, 10, 7, 4, 1, 9];
+
+function averageNumbers(array){
+  for (var suma= 0, i = 0; i < array.length; i++) {
+    suma+= array[i]; 
+  }
+  return suma / array.length;
+}
```

Comment:
Inicializa mejor las variables que vayas a usar en el bloque for fuera de éste. También podías usar reduce( ) para este ejercicio o reutilizar la función anónima 'suma' que hiciste antes. Por lo demás es correcto